snippets/product-variant-options.liquid

```diff
@@ -54,68 +30,81 @@
     else
       assign swatch_value = null
     endif
+
+    assign option_disabled = true
+    if value.variant.available
+      assign option_disabled = false
+    endif
   -%}
 
   {%- capture input_id -%}
     {{ section.id }}-{{ option.position }}-{{ forloop.index0 -}}
   {%- endcapture -%}
 
+  {%- capture input_name -%}
+    {{ option.name }}-{{ option.position }}
+  {%- endcapture -%}
```

Comment:
In a combined listing, parent and child options are surfaced together, so it's technically possible for a product to have multiple options with the same `option.name`. So that the form doesn't break in this case, we not differentiate with position.